api/elections.go

```diff
@@ -281,66 +281,104 @@ func (a *API) electionScrutinyHandler(msg *apirest.APIdata, ctx *httprouter.HTTP
+			continue
+		}
+		intermediateResults = append(intermediateResults, processResult)
+	} // now we have an array of non void results set by a set of oracles via SetProcessResults
+
+	if len(intermediateResults) == 0 {
+		return fmt.Errorf("no results available")
+	}
+
+	// Results are being compared because we want to make sure that
+	// all oracles are sending the same results.
+	equalResults := func(baseResult, otherResult *models.ProcessResult) bool {
+		// if the number of votes is different, the results are different
+		if len(baseResult.Votes) != len(otherResult.Votes) {
+			log.Warnw("different number of votes",
+				map[string]interface{}{
+					"baseResult number of votes":  len(baseResult.Votes),
+					"otherResult number of votes": len(otherResult.Votes),
+				})
+			return false
+		}
+		for k, vote1 := range otherResult.Votes {
+			vote2 := baseResult.Votes[k]
+			if vote1 == nil {
+				log.Warnw("invalid question result (nil)",
+					map[string]interface{}{
+						"question result index": k,
+					})
 				continue
 			}
-			if len(firstResult.Votes) != len(processResult.Votes) {
-				log.Error("votes length do not match")
+			if len(vote1.Question) != len(vote2.Question) {
+				log.Warnw("question length do not match",
+					map[string]interface{}{
+						"baseResult question length":  len(vote2.Question),
+						"otherResult question length": len(vote1.Question),
+					})
 				continue
 			}
-			// if consensus results do not match, return error
-			for k, questionResult := range processResult.Votes {
-				if questionResult == nil {
-					log.Errorf("invalid question result")
-					continue
-				}
-				if len(questionResult.Question) != len(firstResult.Votes[k].Question) {
-					log.Errorf("question length do not match")
+			for kk, question1 := range vote1.Question {
+				question2 := vote2.Question[kk]
+				if question1 == nil {
+					log.Warnw("invalid question option (nil)",
+						map[string]interface{}{
+							"question option index": kk,
+						})
 					continue
 				}
-				for kk, questionOption := range questionResult.Question {
-					if len(questionOption) == 0 {
-						log.Errorf("invalid question option")
-						continue
-					}
-					if len(firstResult.Votes[k].Question[kk]) != len(questionOption) {
-						log.Errorf("question option length do not match")
-						continue
-					}
-					if !bytes.Equal(questionOption, firstResult.Votes[k].Question[kk]) {
-						log.Errorf("election results for signer %s missmatch %s != %s",
-							questionResult.String(),
-							firstResult.Votes[k].String(),
-							common.BytesToAddress(processResult.OracleAddress),
-						)
-						return fmt.Errorf("reported election results missmatch")
-					}
+				if !bytes.Equal(question1, question2) {
+					log.Warnw("question option bytes do not match",
+						map[string]interface{}{
+							"baseResult question option":  fmt.Sprintf("%x", question2),
```

Comment:
So, there is no need to make so explicit the name of the value. On the first parameter you are adding the context message and the next just print useful variable information in key/value format.:

`"question option do not match", "option1", fmt.Sprintf("%x", question2), "option2", fmt.Sprintf("%x", question1))`